src/osd/osd_types.h

```diff
@@ -4420,15 +4420,47 @@ static inline ostream& operator<<(ostream& out, const notify_info_t& n) {
 	     << " " << n.timeout << "s)";
 }
 
+struct chunk_info_t {
+  enum {
+    FLAG_DIRTY = 1, 
+    FLAG_MISSING = 2,
+  };
+  uint64_t length;
```

Comment:
I think we should add a uint32_t offset here too, which is the offset in the target object.  That way teh chunk can reference a fragment of the target object.

Reply:
Fixed